--- /usr/local/stow/mwrender/src/mwlib/lib/python2.7/site-packages/mwlib/net/sapi.py	2017-07-17 23:55:57.765888000 -0500
+++ /usr/local/stow/mwrender/src/mwlib/lib/python2.7/site-packages/mwlib/net/sapi.py	2017-07-18 00:14:49.258460352 -0500
@@ -99,6 +99,14 @@
         url = self._build_url(**kwargs)
         return self._fetch(url)
 
+    def _get(self, **kwargs):
+        args = {'format': 'json'}
+        args.update(**kwargs)
+        getdata = urllib.urlencode(args)
+        req = urllib2.Request(self.apiurl, getdata)
+        res = self._fetch(req)
+        return loads(res)
+
     def _post(self, **kwargs):
         args = {'format': 'json'}
         args.update(**kwargs)
@@ -179,6 +187,12 @@
                 siprop.pop()
         raise RuntimeError("could not get siteinfo")
 
+    def get_token(self):
+       """Fetch and return a login token from the mediawiki server"""
+       args = dict(action="query", meta="tokens", type="login", format="json")
+       r = self._get(**args)
+       return r['query']['tokens']['logintoken']
+
     def login(self, username, password, domain=None, lgtoken=None):
         args = dict(action="login",
                     lgname=username.encode("utf-8"),
@@ -190,6 +204,8 @@
 
         if lgtoken is not None:
             args["lgtoken"] = lgtoken.encode("utf-8")
+        else:
+            args["lgtoken"] = self.get_token().encode("utf-8")
 
         res = self._post(**args)
 
