diff --git a/includes/search/SearchResult.php b/includes/search/SearchResult.php
index 21effbb..3fafebc 100644
--- a/includes/search/SearchResult.php
+++ b/includes/search/SearchResult.php
@@ -147,6 +147,7 @@ class SearchResult {
 		list( $contextlines, $contextchars ) = $this->searchEngine->userHighlightPrefs();
 
 		$h = new SearchHighlighter();
+		$this->mText = strip_tags($this->mText);
 		if ( count( $terms ) > 0 ) {
 			if ( $wgAdvancedSearchHighlighting ) {
 				return $h->highlightText( $this->mText, $terms, $contextlines, $contextchars );
